### PR TITLE
Add a specific error on compressed frames

### PR DESCRIPTION
**Pull request by vbfox**
Instead of the generic 'invalid frame byte' a more specific error `compressed frames aren't supported` is returned when a compressed frame is received

Related to https://github.com/grpc/grpc-web/issues/561

### CODE REVIEW

**Comment by stanley-cheung** [comment]
@vbfox Sorry for the _extreme_ delay. Just got around to run the tests and there was a syntax error

```
/github/grpc-web/packages/grpc-web/node_modules/.bin/google-closure-compiler --js=exports.js --js=../../javascript --js=../../third_party/closure-library --entry_point=grpc.web.Exports --externs=externs.js --dependency_mode=PRUNE --compilation_level=ADVANCED_OPTIMIZATIONS --generate_exports --export_local_property_definitions --js_output_file=index.js
../../javascript/net/grpc/web/grpcwebstreamparser.js:252: ERROR - [JSC_PARSE_ERROR] Parse error. '(' expected
const Parser = GrpcWebStreamParser;
      ^

1 error(s), 0 warning(s)
npm ERR! code ENOENT
npm ERR! syscall access
npm ERR! path /github/grpc-web/packages/grpc-web/node_modules/grpc-web.js
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, access '/github/grpc-web/packages/grpc-web/node_modules/grpc-web.js'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2020-12-05T01_20_58_770Z-debug.log
Service 'prereqs' failed to build: The command '/bin/sh -c cd ./packages/grpc-web &&   npm install &&   npm run build &&   npm link' returned a non-zero code: 254
```

Again my apologies for not getting to this.

File: javascript/net/grpc/web/grpcwebstreamparser.js [hunk 1]
```diff
@@ -186,12 +186,17 @@ class GrpcWebStreamParser {
...
+      if (isCompressed !== 0) {
+        parser.error_(inputBytes, pos, 'compressed frames aren\'t supported');
       } else {
-        parser.error_(inputBytes, pos, 'invalid frame byte');
+        if (frameType == FrameType.DATA) {
+          parser.frame_ = frameType;
+        } else if (frameType == FrameType.TRAILER) {
+          parser.frame_ = frameType;
+        } else {
+          parser.error_(inputBytes, pos, 'invalid frame byte');
```
Comment (stanley-cheung) [inline javascript/net/grpc/web/grpcwebstreamparser.js:199]: There's a missing '}' after this line, looks like.

**Comment by vbfox** [comment]
🤷🏼

**Comment by sampajano** [comment]
Apologies that these PRs slipped through the cracks during team transition.. If you think some of them are still valid or necessary, please feel free to reopen.. sorry!

**Pull request closed without merging**